tests/testsuite/rust_version.rs

```diff
@@ -0,0 +1,143 @@
+//! Tests for targets with `rust-version`.
```

Comment:
I'd like to see the following tests:

- That it is feature gated (see all the *_gated tests).
- How it works with registry dependencies.
- Bad syntax for `rust-version`.

Reply:
Added a `_gated` test and tests for bad syntax. Any suggestion for a registry dependency test I can crib off of?

Reply:
Looking closer, I would just change `rust_version_dependency_fails` to use `Package::new` to create the `bar` dependency instead of using a path dependency. This is just to check the behavior when it is used as a registry dependency. Path and registry dependencies aren't all that different, and I don't expect there to be any problems, just seems nice to verify.

I can't think of any tests that are exactly the same, but [this](https://github.com/rust-lang/cargo/blob/a73e5b7d567c3036b296fc6b33ed52c5edcd882e/tests/testsuite/features_namespaced.rs#L99-L137) is similar. There is some documentation and examples on how to use `Package::new` [here](https://github.com/rust-lang/cargo/blob/a73e5b7d567c3036b296fc6b33ed52c5edcd882e/crates/cargo-test-support/src/registry.rs#L73-L134).

Reply:
Done.